lib/archethic_cache/lru.ex

```diff
@@ -61,6 +61,16 @@ defmodule ArchethicCache.LRU do
      }}
   end
 
+  def code_change(_, state, _) do
+    # As we use hot reload we need to update also the lambda function stored in
+    # permanent term or GenServer state. But as we don't know the function definition
+    # we kill the current GenServer and the Supervisor will restart it with the new function
+    # using the last version
+    me = self()
+    Task.start(fn -> Process.exit(me, :kill) end)
```

Comment:
To avoid this, we should expose MFA instead of a private function in the cache to be called, avoiding the need to kill the process just to make a hot reload.